update/Pos/Update/Poll/Logic/Apply.hs

```diff
@@ -300,27 +304,32 @@ applyImplicitAgreement (flattenSlotId -> slotId) cd hh = do
 -- confirmed or discarded (approved become confirmed, rejected become
 -- discarded).
 applyDepthCheck
-    :: ApplyMode m
+    :: forall m . ApplyMode m
     => EpochIndex -> HeaderHash -> ChainDifficulty -> m ()
 applyDepthCheck epoch hh (ChainDifficulty cd)
     | cd <= blkSecurityParam = pass
     | otherwise = do
         deepProposals <- getDeepProposals (ChainDifficulty (cd - blkSecurityParam))
+        -- 1. Group psoposals by application name
+        -- 2. Sort proposals in each group by tuple
+        --     (decision, whether decision is implicit, positive stake, slot when it has been proposed)
+        -- 3. All proposals in each group except the head we discard
+        -- 4. Concat all groups and process all proposals
```

Comment:
I'd write `Concatenate` instead of `concat`, you're writing comment English.

Reply:
English motherf***er, do you speak it? (c)